code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/identity/TokenResourceManager.java

```diff
@@ -61,6 +66,7 @@ private Token createToken(ApiRequest request) {
             throw new ClientVisibleException(ResponseCodes.SERVICE_UNAVAILABLE,
                     "codeTypeInvalid", "Code type provided invalid.", null);
         }
+        token.setJwt(authTokenDao.createToken(token.getJwt(), token.getAuthProvider(), ((Policy) ApiContext.getContext().getPolicy()).getAccountId()).getKey());
```

Comment:
This still looks wrong since we're calling setJwt() but passing in the JWT key which I don't think is a valid JWT